src/runtime.js

```diff
@@ -342,40 +327,87 @@ function startRegisterOrChunk(global, fnOrStruct, register) {
+    return waitForBodyOpenPromise(global.document).then(() => {
+      // Ensure that all declared extensions are marked and stubbed.
+      stubElementsForDoc(global.AMP.ampdoc);
+    });
+  });
+}
 
-    if (getMode().development) {
-      global.AMP.toggleRuntime = viewer.toggleRuntime.bind(viewer);
-      global.AMP.resources = Services.resourcesForDoc(documentElement);
-    }
+/**
+ * Applies the runtime to a given global scope for a single-doc mode.
+ * Multi frame support is currently incomplete.
+ * @param {!Window} global Global scope to adopt.
+ * @return {!Promise}
+ */
+export function adoptWithMultidocDeps(global) {
+  return adoptShared(global, global => {
+    // Shared runtimes variables between both multi-doc and single-doc pages
+    adoptServicesAndResources(global);
 
-    const viewport = Services.viewportForDoc(documentElement);
-    global.AMP.viewport = {};
-    global.AMP.viewport.getScrollLeft = viewport.getScrollLeft.bind(viewport);
-    global.AMP.viewport.getScrollWidth = viewport.getScrollWidth.bind(viewport);
-    global.AMP.viewport.getWidth = viewport.getWidth.bind(viewport);
+    // Dependencies to the MultiDocManager
+    adoptMultiDocDeps(global);
 
     return waitForBodyOpenPromise(global.document).then(() => {
       // Ensure that all declared extensions are marked and stubbed.
-      stubElementsForDoc(ampdoc);
+      stubElementsForDoc(global.AMP.ampdoc);
     });
   });
 }
 
+/**
+ * Adopt shared runtimes variables between both multi-doc and single-doc pages
+ * @param {!Window} global Global scope to adopt.
+ */
+function adoptServicesAndResources(global) {
+  const {documentElement} = global.document;
+
+  const ampdocService = Services.ampdocServiceFor(global);
+  const ampdoc = ampdocService.getSingleDoc();
+  global.AMP.ampdoc = ampdoc;
+
+  const viewer = Services.viewerForDoc(documentElement);
+  global.AMP.viewer = viewer;
+
+  if (getMode().development) {
+    global.AMP.toggleRuntime = viewer.toggleRuntime.bind(viewer);
+    global.AMP.resources = Services.resourcesForDoc(documentElement);
+  }
+
+  const viewport = Services.viewportForDoc(documentElement);
+  global.AMP.viewport = {};
+  global.AMP.viewport.getScrollLeft = viewport.getScrollLeft.bind(viewport);
+  global.AMP.viewport.getScrollWidth = viewport.getScrollWidth.bind(viewport);
+  global.AMP.viewport.getWidth = viewport.getWidth.bind(viewport);
+}
+
+/**
+ * Adopt MultiDocManager dependencies
+ * @param {!Window} global Global scope to adopt.
+ */
+function adoptMultiDocDeps(global) {
+  global.AMP.installAmpdocServices = installAmpdocServices.bind(null);
+  global.AMP.combinedCss = ampDocCss + ampSharedCss;
+}
+
 /**
  * Applies the runtime to a given global scope for shadow mode.
  * @param {!Window} global Global scope to adopt.
  * @return {!Promise}
  */
 export function adoptShadowMode(global) {
   return adoptShared(global, (global, extensions) => {
+    // shadow mode already adopted
+    if (global.AMP.attachShadowDoc) {
```

Comment:
Where is this set?

Reply:
Hmm not sure I understand, it's set under it on line `426`

Reply:
Ahh, that's not part of Github's diff display.